internal/stream/stream_test.go

```diff
@@ -250,3 +251,150 @@ func TestPushStream_Stop(t *testing.T) {
 	err := ps.Stop()
 	assert.Nil(t, err)
 }
+
+func TestNewPushStream_Failure(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	subscriptionCoreMock := mocks1.NewMockICore(ctrl)
+	subscriberCoreMock := mocks2.NewMockICore(ctrl)
+	workerID := uuid.New().String()
+	httpConfig := &httpclient.Config{}
+	type args struct {
+		ctx              context.Context
+		nodeID           string
+		subName          string
+		subscriptionCore subscription.ICore
+		subscriberCore   subscriber.ICore
+		config           *httpclient.Config
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *PushStream
+		wantErr bool
+	}{
+		{
+			name: "New Push Stream Failure",
+			args: args{
+				ctx:              ctx,
+				nodeID:           workerID,
+				subName:          subName,
+				subscriptionCore: subscriptionCoreMock,
+				subscriberCore:   subscriberCoreMock,
+				config:           httpConfig,
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			subscriptionCoreMock.EXPECT().Get(gomock.Any(), subName).DoAndReturn(
+				func(arg0 context.Context, arg1 string) (*subscription.Model, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return getMockSubModel(""), nil
+				},
+			)
+			got, err := NewPushStream(tt.args.ctx, tt.args.nodeID, tt.args.subName, tt.args.subscriptionCore, tt.args.subscriberCore, tt.args.config)
+			if err == nil {
```

Comment:
if you are expecting an error, then why this check here..

Reply:
removed